src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -374,6 +380,672 @@ static void appendModifiers(ImmutableArray<CustomModifier> customModifiers, Stri
+
+                for (int i = 0; i < attributesBuilder.Count; i++)
+                {
+                    builder.Append('[');
+                    builder.Append(attributesBuilder[i]);
+                    builder.Append("] ");
+                }
+
+                attributesBuilder.Free();
+            }
+
+            static void appendAttribute(CSharpAttributeData attribute, StringBuilder builder)
+            {
+                Debug.Assert(attribute.AttributeClass is not null);
+                appendType(attribute.AttributeClass, builder);
+
+                if (attribute.CommonConstructorArguments.IsEmpty && attribute.CommonNamedArguments.IsEmpty)
+                {
+                    return;
+                }
+
+                builder.Append('(');
+                bool needComma = false;
+                foreach (TypedConstant argument in attribute.CommonConstructorArguments)
+                {
+                    if (needComma)
+                    {
+                        builder.Append(", ");
+                    }
+
+                    appendAttributeArgument(argument, builder);
+                    needComma = true;
+                }
+
+                if (!attribute.CommonNamedArguments.IsEmpty)
+                {
+                    var namedArgumentsBuilder = ArrayBuilder<string>.GetInstance(attribute.CommonNamedArguments.Length);
+                    foreach (KeyValuePair<string, TypedConstant> namedArgument in attribute.CommonNamedArguments)
+                    {
+                        var stringBuilder = PooledStringBuilder.GetInstance();
+                        appendAttributeNamedArgument(namedArgument, stringBuilder.Builder);
+                        namedArgumentsBuilder.Add(stringBuilder.ToStringAndFree());
+                    }
+
+                    namedArgumentsBuilder.Sort(StringComparer.Ordinal); // Actual order doesn't matter - just want to be deterministic
+
+                    foreach (string namedArgument in namedArgumentsBuilder)
+                    {
+                        if (needComma)
+                        {
+                            builder.Append(", ");
+                        }
+
+                        builder.Append(namedArgument);
+                        needComma = true;
+                    }
+
+                    namedArgumentsBuilder.Free();
+                }
+
+                builder.Append(')');
+            }
+
+            static void appendAttributeArgument(TypedConstant argument, StringBuilder builder)
+            {
+                if (argument.Kind == TypedConstantKind.Error)
+                {
+                    builder.Append("error");
+                    return;
+                }
+
+                if (argument.IsNull)
+                {
+                    Debug.Assert(argument.Type is not null);
+                    builder.Append('(');
+                    appendType(((PublicModel.TypeSymbol)argument.Type).UnderlyingTypeSymbol, builder);
```

Comment:
>Type [](http://example.com/codeflow?start=65&length=4)

If this type is coming from attribute signature, then we cannot make any assumption about the lack of custom modifiers. Also, we probably do not care about any C#-ism in the types from attributes, otherwise the name start depending on an insignificant information that can come from a different assembly.